unit/java_bytecode/java_types/generic_type_index.cpp

```diff
@@ -0,0 +1,33 @@
+/*******************************************************************\
+
+ Module: Unit tests for java_types
+
+ Author: Diffblue Ltd.
+
+\*******************************************************************/
+
+#include <testing-utils/catch.hpp>
+#include <java_types.h>
+
+SCENARIO("generic_type_index", "[core][java_types]")
+{
+  // Arrange
+  const auto symbol_type = symbol_typet("MyType");
+  const auto generic_symbol_type = java_generic_symbol_typet(
+    symbol_type, "LGenericClass<TX;TY;>;", "PrefixClassName");
```

Comment:
Suggested additions: Non-single letter generic parameter name, name corresponding to a real class